features/jdbc-collector/src/main/java/org/opennms/netmgt/collectd/JdbcCollector.java

```diff
@@ -241,8 +241,11 @@ public CollectionSet collect(CollectionAgent agent, EventProxy eproxy, Map<Strin
                             JdbcCollectionResource resource = null;
                             
                             // Create the appropriate resource container.
-                            if(singleInstance) {
-                                resource = nodeResource;
+                            if(null==query.getResourceType()) {
+                                if (!singleInstance){
+                                    LOG.warn("Multiple row results but there is no resourceType defined : only first row will be persisted");
+                                    resource = nodeResource;
+                                }
```

Comment:
Won't resource be `null` if resource type is defined, and this is a single instance?